text/XXXX-full-serialisation.md

```diff
@@ -0,0 +1,140 @@
+- Start Date: 2017-03-27
+- RFC PR: (leave this empty)
+- Refract Issue: (leave this empty)
+
+# Refract Full JSON Serialisation
+
+## Summary
+
+Full Refract serialisation is not clearly defined in the Refract specification
+and there are a few ambiguities that may occur when parsing a Refract document.
+
+## Motivation
+
+It should be clear to consumers how to serialise and de-serialise refracted
+elements.
+
+At the moment there are no formal rules for full serialisation and some
+implementations are inconsistent. Many Refract serialisers support embedding
+both refracted and raw JSON data in a Refract document, however many
+de-serialisers and consumers of Refract do not support both forms in all
+places.
+
+Since an element does not need to be refracted, that leads to unrefracted
+elements being used in some cases causing ambiguity and leads to consumers
+having to know details about a Refract namespace to be able to truely parse it.
+
+It should be possible for a Refract consumer to consume a Refract document
+without knowing about specific elements from a namespace. There should be no
+ambiguities when parsing a Refract document.
+
+## Detailed design
+
+We should provide a full specification on how Refract elements can be
+serailised as JSON and recommendations on how to consume Refract elements in
+JSON.
+
+### JSON Serialisation
+
+A Refract element MUST be serialised as JSON as follows:
+
+- element (required, string)
+- meta (optional)
+- attributes (optional)
+- content (optional)
+
+The name of the element is always required, however other JSON values may be
+omitted, for example if there is no meta keys then it SHOULD be omitted from
+the JSON object.
+
+#### Content
+
+The content inside JSON Serialisation MUST always be a Refracted element, an
+array of Refracted element or a primitive type such as string, number, boolean
+or none. The only exceptions to this rule is for any element types described
```

Comment:
I do not think Link Element actually would be exclusions to this rule.

**Link Element** does not use content and does not have direct content at all. It only supports `href` and `relation` as ATTRIBUTES not inside content.

** Element Pointer** on the other hand does use an enum where the value may be an object, but I wonder if we can simplify the element pointer value to be a string element with an attribute instead.

Currently the following is defined in the spec for the value for Element Pointer:

> #### Members
> 
> - (string) - A URL to an ID of an element in the current document
> - (object)  - The URL and path of the referenced element
>     - `href` (string, required) - URL or ID of element
>     - `path` (enum) - Path of referenced element to transclude instead of element itself
>         - meta - The meta data of the referenced element
>         - attributes - The attributes of the referenced element
>         - content - The content of the referenced element

Perhaps this could be simplified to:

```apib
+ attributes:
    + path (enum, optional)
        + meta - The meta data of the referenced element
        + attributes - The attributes of the referenced element
        + content - The content of the referenced element
+ content (string)
```

That way, the only "special-cased" is member, which we can say the value is a "key-value pair" as @smizell suggests.